blocks/subscriptions-block/components/usePaymentOptions.jsx

```diff
@@ -1,65 +1,76 @@
 import { useState, useEffect } from "react";
 
-import useSales from "./useSales";
+import { useSales } from "@wpmedia/arc-themes-components";
 
 const STRIPE_PAYMENT_METHOD_ID = 18;
 const PAYPAL_METHOD_ID = 10;
 
-const getPaymentMethodByID = (paymentOptions, paymentMethodTypeID, paymentMethodID) => {
+export const getPaymentMethodByID = (paymentOptions, paymentMethodTypeID, paymentMethodID) => {
 	const stripeDefault = paymentOptions?.find(
 		(opts) =>
 			opts?.paymentMethodType === paymentMethodTypeID && opts?.paymentMethodID === paymentMethodID,
 	);
 
 	if (stripeDefault) {
 		return stripeDefault;
-	} else {
+	}
+
+	if (!stripeDefault) {
 		const allStripeIntents = paymentOptions?.filter(
 			(opts) => opts?.paymentMethodType === paymentMethodTypeID,
 		);
 
-		var minStripeIntentsID = allStripeIntents.reduce(function (res, obj) {
-			return obj.paymentMethodID < res.paymentMethodID ? obj : res;
-		});
-		return minStripeIntentsID;
+		return allStripeIntents.reduce((res, obj) =>
+			obj.paymentMethodID < res.paymentMethodID ? obj : res,
+		);
 	}
+
+	return undefined;
 };
 
 // On subscriptions side we can setup multiple payment providers with the same typeID.
 // This hook is on charge to return the default one or the one with the min paymentMethodID
-export const usePaymentOptions = (stripeIntentsDefaultID, paypalDefaultID) => {
+const usePaymentOptions = (stripeIntentsDefaultID, paypalDefaultID) => {
 	const [paymentOpts, setPaymentOpts] = useState();
 	const [paypal, setPaypal] = useState();
 	const [stripeIntents, setStripeIntents] = useState();
+	const [isFetching, setIsFetching] = useState(true);
 	const [error, setError] = useState();
 
 	const { Sales } = useSales();
 
+	const fetchData = async () => {
+		try {
+			const options = await Sales.getPaymentOptions();
+			const stripeSettings = getPaymentMethodByID(
+				options,
+				STRIPE_PAYMENT_METHOD_ID,
+				stripeIntentsDefaultID,
+			);
+			setStripeIntents(stripeSettings);
+
+			const paypalSettings = getPaymentMethodByID(options, PAYPAL_METHOD_ID, paypalDefaultID);
+			setPaypal(paypalSettings);
+			
+			setPaymentOpts(options);
+			setIsFetching(false);
+		} catch (e) {
+			setError(e);
+			setIsFetching(false);
+		}
+	};
+
 	useEffect(() => {
-		const fetchData = async () => {
-			try {
-				const options = await Sales.getPaymentOptions();
-				const stripe = getPaymentMethodByID(
-					options,
-					STRIPE_PAYMENT_METHOD_ID,
-					stripeIntentsDefaultID,
-				);
-				const paypal = getPaymentMethodByID(options, PAYPAL_METHOD_ID, paypalDefaultID);
-                setPaymentOpts(options);
-                setPaypal(paypal);
-                setStripeIntents(stripe);
-			} catch (e) {
-				setError(e);
-			}
-		};
 		fetchData();
+		// eslint-disable-next-line
```

Comment:
Why are we disabling this?  The dependencies should be filled out.

Reply:
Oddly, though... this one may be better served to put that function back in the `useEffect` because the fetch function is created each render and that will trigger the effect.  If you only want this to run once, the function isn't necessary, but I assume you want it to run each time the `STRIPE_PAYMENT_METHOD_ID` or `PAYPAL_METHOD_ID` changes (which should be never/once)

Reply:
I imagine this should be:
```
	useEffect(() => {
		const fetchData = async () => {
			try {
				const options = await Sales.getPaymentOptions();
				const stripeSettings = getPaymentMethodByID(
					options,
					STRIPE_PAYMENT_METHOD_ID,
					stripeIntentsDefaultID,
				);
				setStripeIntents(stripeSettings);

				const paypalSettings = getPaymentMethodByID(options, PAYPAL_METHOD_ID, paypalDefaultID);
				setPaypal(paypalSettings);

				setPaymentOpts(options);
				setIsFetching(false);
			} catch (e) {
				setError(e);
				setIsFetching(false);
			}
		};
		fetchData();
	}, [STRIPE_PAYMENT_METHOD_ID, PAYPAL_METHOD_ID]);
```

Reply:
If the fetch is outside of the `useEffect`, it will be recreated on each render.